pkg/openstack/barbican.go

```diff
@@ -142,6 +149,9 @@ func ReconcileBarbican(ctx context.Context, instance *corev1beta1.OpenStackContr
 			condition.SeverityInfo,
 			corev1beta1.OpenStackControlPlaneBarbicanReadyRunningMessage))
 	}
+	instance.Status.ContainerImages.BarbicanAPIImage = version.Status.ContainerImages.BarbicanAPIImage
```

Comment:
I guess we will check the observedGeneration on barbican along with IsReady before setting the images in our status as soon as observedGeneration field is available